### ISSUE TITLE
node 19 hangs when installing Realm

### ISSUE THREAD

**Issue by kneth** [realm/realm-js#5136]
### How frequently does the bug occur?

Sometimes

### Description

Using node 19, `npm install realm` seems to hang.

### Can you reproduce the bug?

Not yet

### Reproduction Steps

```sh
nvm use 19
mkdir foobar
cd foobar
npm init -y
npm install realm
```

### Version

11.2.0

### What SDK flavour are you using?

Local Database only

### Are you using encryption?

No, not using encryption

### Platform OS and version(s)

MacOS, Linux

**Comment by kneth** [comment]
I am not able to reproduce it with the following (MacOS 12.6.1 - Apple Silicon):

* node 19.1.0, npm 8.19.3
* node 19.1.0, npm 9.1.2
* node 19.1.0, yarn 1.22.19

**Comment by zoobibackups** [comment]
this issue was with macOS Monterey 12.4 and latest node version while install in react-native. after downgrade to node v18.2.0 I installed it successfully.

**Comment by LionStoneSoft** [comment]
> this issue was with macOS Monterey 12.4 and latest node version while install in react-native. after downgrade to node v18.2.0 I installed it successfully.

Yep, same here. I couldn't npm install realm on node 19.2, downgrading to 18.2 worked. Thanks.

**Comment by kneth** [comment]
I will reopen the issue and allocate some time to investigate

**Comment by johnbonds** [comment]
Just wanted to leave a comment here that we are seeing this as well. We're using yarn. Did a clean complete reinstall of the OS and when we run a yarn install on our RN mobile app, it hangs on "realm". Downgrading to node 18.2 seems to work.

**Comment by shichunyu** [comment]
Trying to do `npm install realm` for an Electron App with React and also hangs on `reify:realm:`

**Comment by soorajsprakash** [comment]
I'm on v19.3.0, and can confirm that it hangs on `reify:realm`.

**Comment by cyril36** [comment]
I am using the docker image node:19.1.0 and i have the exact same issue

**Comment by Pointotech** [comment]
Same issue with Node 19.4.0 and Yarn 1.22.19 on MacOS 11.7.1.

**Comment by kneth** [comment]
I am not able to reproduce it on my laptop (MacOS 12.6.2, M1) with node 19.4.0 and yarn 1.22.19. I have manually checked the downloaded binary (`node_modules/realm/build/Release/realm.node`) to verify it exists and is a shared library for ARM64.

**Comment by BruceSuperProgramer** [comment]
Having same issue but it fixed after I removed node 19.4.0 and installed node 14.21.2

**Comment by jataca** [comment]
Still an issue with node 19.5 on Windows 10.

**Comment by justshowkat** [comment]
I am using node 19.3.0 and getting stuck at **npm install realm**

this is the line that I am getting stuck at:
```
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
(##################) ⠋ reify:realm: timing metavuln:cache:get:security-advisory:babel-preset-expo:C8YGR8HMy4cEwxDrWMFcv+e2Uv5RTnK
```

**Comment by kneth** [comment]
As I am not able to reproduce it, I would like to know more about which versions, you are using (npm, etc.)

**Comment by ElicaInc** [comment]
faced the same issue on node 19, but works fine on node v18.14.2

**Comment by GitMurf** [comment]
As documented here (https://github.com/realm/realm-js/issues/5547#issuecomment-1458517752) I am running into this on Windows 11 with Node 19.7 and NPM 9.6 (also happened with NPM 8). It seems that it may be hanging on the step for "submit-analytics.js" 🤷‍♂️

**Comment by DenisLebedinsky** [comment]
I have same issue with node 19.3.0. Installing 18.15.0 fixed it

**Comment by kneth** [comment]
I have a fix in https://github.com/realm/realm-js/pull/5554

**Comment by kneth** [comment]
Released in v11.5.2
